serverlessworkflow/modules/ROOT/pages/eventing/event-correlation-with-workflows.adoc

```diff
@@ -1 +1,205 @@
+
+The given example uses Callback States, such as the  <<ref-validade-user-email-state,`Validate User Email`>>, indicating once the workflow execution achieves this state, it publishes an event of type <<ref-correlation-event-definition,`validateAccountEmailEvent`>> and wait to receive an event of type <<ref-correlation-event-definition,`validatedAccountEmailEvent`>>. For more information on xref:eventing/working-with-callbacks.adoc[Working with callbacks].
+
+[[ref-validade-user-email-state]]
+.Callback State Definition
+[source,json]
+----
+{
+  "name": "Validate User Email",
+  "type": "callback",
+  "action": {
+    "name": "publish validate event",
+    "eventRef": {
+      "triggerEventRef": "validateAccountEmailEvent"
+    }
+  },
+  "eventRef": "validatedAccountEmailEvent",
+  "transition": "Activate User Account"
+}
+----
+
+[IMPORTANT]
+The produced events contain the same correlation attributes set once the workflow was created.
+
+[[ref-validade-user-email-event]]
+.Produced Callback State event `validateAccountEmailEvent`
+[source,json]
+----
+{
+  "id": "7640a0af-b7fb-4d94-9d9d-3aa1ace60e79",
+  "source": "/process/correlation",
+  "type": "validateAccountEmail",
+  "time": "2022-07-25T16:22:53.735128049-03:00",
+  "data": {
+    "email": "test@test.com",
+    "userId": "12345"
+  },
+  "specversion": "1.0",
+  "kogitoprocinstanceid": "69019826-daef-4fb4-880b-c1658c4e49bc",
+  "kogitoprocid": "correlation",
+  "kogitoprocversion": "1.0",
+  "kogitousertaskist": "1",
+  "kogitoproctype": "SW",
+  "userid": "12345"
+}
+----
+
+All consumed events must contain the same correlation attributes since they are used to identify the workflow instance, in this example
+`userid` with value `12345`, see <<ref-validaded-user-email-event,validatedAccountEmailEvent>>.
+
+[[ref-validaded-user-email-event]]
+.Consumed Callback State event `validatedAccountEmailEvent`
+[source,json]
+----
+{
+  "specversion": "1.0",
+  "id": "953f07a7-aea8-4956-8775-85ab59366fe6",
+  "source": "",
+  "type": "validatedAccountEmail",
+  "time": "2022-07-25T16:29:27.320408379-03:00",
+  "userid": "12345",
+  "data": null
+}
+----
+
+== Correlation configuration
+
+The workflow engine stores the correlation information in the same persistence mechanism configured in the application. If no persistence addon is configured, the correlation information is stored in memory, this means,  all correlation information is lost upon the application restart, so it should only be used for testing purposes. For more information about persistence configuration, see xref:persistence/persistence-with-postgresql.adoc[Persistence with PostgresSQL databases].
+
+[NOTE]
+====
+Currently, the only persistency addon which supports correlation is the `kogito-addons-quarkus-persistence-jdbc` configured for PostgreSQL. In the next releases, other persistency addons will be supported.
+====
```

Comment:
Can you please include the "Found an issue" link and the "Additional resources" section?

Reply:
+1